geps/gep-2659.md

```diff
@@ -0,0 +1,199 @@
+relationship (or any other, in fact) between GEPs at the moment.
+
+With these previous two changes, the addition of a metadata YAML file, similar
+to what the KEP process uses, seems like it is increasingly necessary. This GEP
+introduces a schema, which will be detailed in `.go` files rather than completely
+in this document.
+
+Lastly, I think that we should clarify our use of [RFC2119](https://www.rfc-editor.org/rfc/rfc2119.txt)
+language - we use MUST, SHOULD, MAY and so on as per that RFC in general,
+but there is an extension in [RFC8174](https://www.rfc-editor.org/rfc/rfc8174.txt)
+that adds that these words are only to be interpreted per the RFC when they are
+in ALL CAPS, and "must" "should", "may" are to be interpreted in their usual
+English meaning (which is not as strong as the RFC2119 one). This seems like
+a _very_ good idea to adopt to me.
+
+## Proposals
+
+### Adopt the RFC8174 modification to RFC2119 language
+
+RFC8174 clarifies that the reserved words MUST be in all-caps to have their
+assigned meaning. This should make the spec clearer over time as we migrate.
+
+### Addition of the concept of GEP "types", and a new GEP Type
+
+We will introduce the concept of GEP types.
+
+#### Types of GEP
+There will be two types of GEP. The existing type will be called `API` GEPs, and
+a new type `Memorandum` (after "Memorandum of Understanding") will be created.
+
+##### API GEPs
+
+GEPs that do not have an assigned type will be considered `API` GEPs. (For
+legacy GEPs). New GEPs created after this GEP is merged will default to the `API`
+type.
+
+As they do currently, these GEPs lay out a problem the API needs to address,
+along with a rationale and a solution for that problem.
+
+This also uses the existing GEP process and phases:
+
+1. Discuss
+1. Provisional
+  1. (optional) Prototyping
+1. Implementable
+1. Experimental
+1. Standard
+1. Completed
+
+As part of the addition of relationships, we will extend these phases, but see
+the "Addition of GEP relationships" section for details, but in short, when an
+API GEP has been Obsoleted, it will be moved to a new "Obsolete" phase (which
+is functionally the same as "Declined").
+
+##### Memorandum GEPs
+
+These GEPs record an agreement on either the definition of a problem, its scope
+and a common language, or further process changes to the GEP process itself.
+The defining characteristic here is that the GEP MUST NOT result in any changes
+to the Gateway API spec, and SHOULD result in further GEPs to further clarify.
+For GEPs that _do_ make changes to the API, but also require further GEPs to
+clarify, they SHOULD use the new "Updated By" relationship instead.
+
+Memorandum GEPs should be used sparingly, and should form the umbrella for a
+significant amount of work, particularly work that may have parts that can
+move through the GEP phases at different speeds.
+
+The process for Memorandum GEPs is much more simple, and has really only one
+main phase - "Accepted".
```

Comment:
Similar to my comment above, if there is only one phase, maybe "Memorandum" is the phase and we don't need to bother with both phase and type here.